sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp

```diff
@@ -258,6 +258,52 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     FileSystemAccessConditions AccessConditions;
   };
 
+  /**
+   * @brief Optional parameters for DataLakeFileSystemClient::RenameDirectorySinglePage
+   * @remark Some optional parameter is mandatory in certain combination.
+   *         More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create
+   */
+  struct RenameDataLakeDirectorySinglePageOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief When renaming a directory, the number of paths that are renamed with each
+     *        invocation is limited. If the number of paths to be renamed exceeds this limit,
+     *        a continuation token is returned in this response header. When a continuation token
+     *        is returned in the response, it must be specified in a subsequent invocation of the
+     *        rename operation to continue renaming the directory.
+     */
+    Azure::Core::Nullable<std::string> ContinuationToken;
+
+    /**
+     * @brief This parameter determines the behavior of the rename operation. The value must be
+     *        PathRenameMode::Legacy or PathRenameMode::Posix, and the default value will be
+     *        PathRenameMode::Posix.
+     */
+    Models::PathRenameMode Mode = Models::PathRenameMode::Posix;
```

Comment:
you may want to reconsider the default value given what Sean said https://github.com/Azure/azure-sdk-for-cpp/pull/1493#issuecomment-768761597

Reply:
I would recommend removing this parameter entirely.  We can add it back in the future if customers ask for it, but that has not been the case on the other platforms.

Reply:
> I would recommend removing this parameter entirely. We can add it back in the future if customers ask for it, but that has not been the case on the other platforms.


I agree

Reply:
Removed.